### DISCUSSION TITLE
Changing contents of <head> in a new Fresh app

### DISCUSSION THREAD

**Discussion by adiabatic** [freshframework/fresh#307] [Q&A] [answered]
Hi!

I installed Fresh just now by running `deno run -r https://fresh.deno.dev .` in an empty directory and let is have every permission that it asked for. When it asked me if I wanted to use `twind.dev`, I said "no". I ran the local server and it seems pretty neat.

However, after looking through the files installed and through all the docs on https://fresh.deno.dev/docs/introduction and a quick look through most pages on https://preactjs.com/guide/v10/getting-started, I couldn't find out how to change the HTML skeleton generated. Sure, I could change most things on the index page by editing `routes/index.tsx`, but I couldn't find out how to add a line like `<link rel="stylesheet" href="/static/all.css" />` or change the `lang` attribute of the `html` element.

How would I do that?

**Comment by Luca00711** [comment, marked as answer]
I have the answer for you, I was looking for the same and I found some.

You can do:
`import { Head } from "$fresh/runtime.ts";`

and use then in `return()`:
```
<Head>
  <link rel="stylesheet" href="/static/all.css" />
</Head>
```

**Reply by lucacasonato** [reply]
Also see https://github.com/denoland/fresh/discussions/301

**Reply by marshellwu** [reply]
but that only change a page, how to apply it globally?

**Reply by nikableh** [reply]
> but that only change a page, how to apply it globally?

You can create your own `<Head/>` component (something like `<MyHead/>`) that will contain `<Head/>` from `$fresh/runtime.ts` with all the tags you want to use. Then just include `<MyHead/>` everywhere you want `<head>` contents to be applied. I think this is much better than applying such settings globally, since it's much more flexible.

**Reply by adiabatic** [reply]
> but that only change a page, how to apply it globally?

Another option: Make a higher-order component.

This higher-order component will have in it:

- `<Head>…</Head>`
- Whatever the other component passed into it outputs

Then, at the bottom of your page-level components, do

```ts

function HomePage(props: HomePageProps) {/* … */};

export default AddingUsualHeadThings(HomePage);
```

**Reply by iamsahilsonawane** [reply]
Using this cannot find the path of the CSS file, there's indeed a file called main.css in the static, but still cannot see the CSS effects

routes/index.tsx
```html
<Head>
    <link rel="stylesheet" href="/static/main.css"></link>
</Head>
```

static/main.css
```css
html {
    color: blue;
    background-color: black;
}
```

Result: It cannot find the path of the CSS file (as checked in the source) and therefore doesn't apply the color styles on the page.

**Reply by germancgn** [reply]
You don't have to include `/static` in href. Try this:
```html
<Head>
    <link rel="stylesheet" href="/main.css"></link>
</Head>
```

**Reply by iamsahilsonawane** [reply]
> You don't have to include `/static` in href. Try this:
> 
> ```
> <Head>
>     <link rel="stylesheet" href="/main.css"></link>
> </Head>
> ```

Yes it's good now thanks!

**Comment by PhoneDroid** [comment]
@lucacasonato Please close, covered by docs.
( https://fresh.deno.dev/docs/examples/modifying-the-head )
( Including correct usage of static assets )
